pkg/rpc/request/param.go

```diff
@@ -210,20 +211,46 @@ func (p *Param) GetIntStrict() (int, error) {
 	if p.IsNull() {
 		return 0, errNotAnInt
 	}
-	if p.cache == nil {
-		var i int
-		err := json.Unmarshal(p.RawMessage, &i)
-		if err != nil {
-			return i, errNotAnInt
-		}
-		p.cache = i
+	value, err := p.fillIntCache()
+	if err != nil {
+		return 0, err
 	}
-	if i, ok := p.cache.(int); ok {
-		return i, nil
+	if i, ok := value.(int64); ok && i == int64(int(i)) {
+		return int(i), nil
 	}
 	return 0, errNotAnInt
 }
 
+func (p *Param) fillIntCache() (interface{}, error) {
```

Comment:
Maybe call it `fillCache`? It fills not only int cache, but also cache for other possible types.

Reply:
It is used only in `Get*Int`, for other types the unmarshaling order is different.